app/src/main/java/com/codesoom/assignment/service/TaskService.java

```diff
@@ -0,0 +1,77 @@
+package com.codesoom.assignment.service;
+
+import com.codesoom.assignment.exception.TaskNotFoundException;
+import com.codesoom.assignment.model.Task;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+@Service
+public class TaskService {
+
+    private Map<Long, Task> tasks = new LinkedHashMap<Long, Task>();
+    private Long newId = 0L;
+
+    public List<Task> list() {
+
+        return new ArrayList<Task>(tasks.values());
+    }
+
+    public Task create(Task task) {
+        Long id = generateId();
+        task.setId(id);
+        task.setState("yet");
+        tasks.put(id, task);
+
+        return task;
+    }
+
+    public Optional<Task> view(Long id) {
+        Optional<Task> task = findTask(id);
+        if (!task.isPresent()) {
+            throw new TaskNotFoundException("요청하신 " + id + "의 Task가 없습니다.");
+        }
+        return task;
+    }
+
+    public Optional<Task> update(Task body, Long id) {
+        Optional<Task> task = findTask(id);
+        if (!task.isPresent()) {
+            throw new TaskNotFoundException("요청하신 " + id + "의 Task가 없습니다.");
```

Comment:
예외를 통한 에러 메시지는 고객에게 주는 것이 아니라 이 코드를 디버깅할 미래의 자기 자신에게 보여주는 거라고 생각하고 작성해 보세요. 지금은 "요청하신 id의 Task가 없습니다."로 전부 통일되어 있군요. 그런데 나중에 문제가 생겨서 로그 파일을 열어봤는데 전부 이 메시지만 있다고 생각해 봅시다. 

```suggestion
            throw new TaskNotFoundException(id + "에 해당하는 Task가 없으므로 업데이트를 할 수 없습니다.");
```

이렇게 이 문제 때문에 어떤 작업을 할 수 없게 되었다는 것까지 표현해 주면 12시까지 야근할 것을 10시까지 야근해도 해결할 수 있을지도 모릅니다. 본인이 디버깅하게 되는 경우 말고도, 나는 휴가를 갔는데 동료들이 이 문제를 해결해야 하는 상황에 놓였다고 생각해 보세요. 에러 메시지가 다 똑같고 원인과 결과를 명백히 설명하지 않는다면 동료들이 고생하게 될 지도 모릅니다.

Reply:
에러 메세지가 고객에게 보여주는 것이라고 생각을 했네요. 
메서드에 맞게 원인과 결과를 명백히 설명하는 메세지로 바꿔야겠네요!